src/tracker/inflights.rs

```diff
@@ -37,13 +42,38 @@ impl Inflights {
             start: 0,
             count: 0,
             cap,
+            incoming_cap: None,
+        }
+    }
+
+    /// Adjust inflight buffer capacity. Set it to `0` will disable the progress.
+    pub fn set_cap(&mut self, incoming_cap: usize) {
+        match self.cap.cmp(&incoming_cap) {
+            Ordering::Equal => self.incoming_cap = None,
+            Ordering::Less => {
+                self.buffer.reserve(incoming_cap - self.count);
+                self.cap = incoming_cap;
+                self.incoming_cap = None;
+            }
+            Ordering::Greater => {
+                if self.count == 0 {
```

Comment:
It seems it has to allocate in the end. So it's simpler to just do allocation now instead of adding a new field.

Reply:
The new field is `incoming_cap`? I think it's required. Please consider calling `set_cap(x)` when `self.count > x`.